flink-clients/src/main/java/org/apache/flink/client/program/rest/UrlPrefixDecorator.java

```diff
@@ -41,7 +40,7 @@
  */
 public class UrlPrefixDecorator<
                 R extends RequestBody, P extends ResponseBody, M extends MessageParameters>
-        implements SqlGatewayMessageHeaders<R, P, M> {
```

Comment:
Where/how is `SqlGatewayMessageHeaders` used after this change?

Reply:
Yes, after looking at the current code it isn't clear to me why it implemented that interface in first place. So you were able to remove it as it has no effect and nothing needs to be done to replace that interface usage?

Reply:
@afedulov can you please chime in?

Reply:
`SQLGatewayUrlPrefixDecorator`, the subclass of `UrlPrefixDecorator`, overrides `getSupportedAPIVersions`and does what `SqlGatewayMessageHeaders::getSupportedAPIVersions` was doing.